Description: s390x: protvirt: Fix stray error_report_err in s390_machine_prot
In case the protection of the machine fails at s390_pv_vm_enable(),
we'll currently report the local_error variable. Problem is that
there's no migration blocker error that we can report at this point so
the pointer is always NULL which leads to a SEGFAULT.

Let's remove the error report.

Signed-off-by: Janosch Frank <address@hidden>
Reported-by: Marc Hartmayer <address@hidden>
Fixes: 520935eedf941da3 ("s390x: protvirt: Add migration blocker")

Author: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Origin: upstream, https://lists.nongnu.org/archive/html/qemu-devel/2020-03/msg07969.html
Bug-Ubuntu: https://bugs.launchpad.net/bugs/<todo-bug>
Last-Update: 2020-03-26
--- a/hw/s390x/s390-virtio-ccw.c
+++ b/hw/s390x/s390-virtio-ccw.c
@@ -360,7 +360,6 @@ static int s390_machine_protect(S390CcwM
     rc = s390_pv_vm_enable();
     if (rc) {
         qemu_balloon_inhibit(false);
-        error_report_err(local_err);
         migrate_del_blocker(pv_mig_blocker);
         error_free_or_abort(&pv_mig_blocker);
         return rc;
